src/Files.App/Data/Factories/ShellContextFlyoutHelper.cs

```diff
@@ -359,20 +359,27 @@ async Task InvokeShellMenuItemAsync(ContextMenu contextMenu, object? tag)
 					itemContextMenuFlyout.SecondaryCommands.Insert(1, sendToItems.FirstOrDefault());
 				}
 
-				// Add items to shell submenu
-				shellMenuItems.Where(x => x.LoadSubMenuAction is not null).ForEach(async x =>
-				{
-					await x.LoadSubMenuAction();
 
+				var itemsWithSubMenu = shellMenuItems.Where(x => x.LoadSubMenuAction is not null);
+				
+				//The order here may be important
+				foreach (var item in itemsWithSubMenu)
+				{
+					await item.LoadSubMenuAction();
```

Comment:
The order is important for `AddItemsToMainMenu` and `AddItemsToOverflowMenu`, but not for `LoadSubMenuAction`.